src/editor/gridstack/gridstackLayout.ts

```diff
@@ -0,0 +1,196 @@
+import { Layout, Widget } from '@lumino/widgets';
+
+import { IIterator, ArrayIterator } from '@lumino/algorithm';
+
+import { Signal } from '@lumino/signaling';
+
+import {
+  GridStack,
+  GridHTMLElement,
+  GridStackNode,
+  GridItemHTMLElement
+} from 'gridstack';
+
+import 'gridstack/dist/gridstack.css';
+
+import { GridStackItem } from './gridstackItemWidget';
+
+import { DashboardView, DashboardCellView } from './../format';
+
+export class GridStackLayout extends Layout {
+  constructor(info: DashboardView) {
+    super();
+    this._margin = info.cellMargin;
+    this._cellHeight = info.cellHeight;
+    this._columns = info.numColumns;
+
+    this._gridItems = new Array<GridStackItem>();
+
+    this.gridItemChanged = new Signal<this, GridStackNode[]>(this);
+  }
+
+  readonly gridItemChanged: Signal<this, GridStackNode[]>;
+
+  dispose(): void {
+    this._gridItems = null;
+    this._grid?.destroy();
+    this._grid = null;
+    super.dispose();
+  }
+
+  onResize(): void {
+    if (this._grid) {
+      this._grid.onParentResize();
+    }
+  }
+
+  onFitRequest(): void {
+    if (this._grid) {
+      this._grid.onParentResize();
+    }
+  }
+
+  iter(): IIterator<Widget> {
+    return new ArrayIterator(this._gridItems);
+  }
+
+  removeWidget(widget: Widget): void {
+    return;
+  }
```

Comment:
Should this method be kept?

Reply:
It's mandatory when extending from Layout.

Reply:
Ah right, thanks!